config/config_test.go

```diff
@@ -32,6 +33,80 @@ func TestIsStack(t *testing.T) {
 	assert.IsTrue(t, !isStack(cfg, "/stack/subdir"))
 }
 
+func TestTryLoadConfig(t *testing.T) {
+	t.Parallel()
+	type want struct {
+		cfgpath string
+		found   bool
+	}
+	type testcase struct {
+		name    string
+		layout  []string
+		fromdir string
+		want    want
+	}
+
+	for _, tc := range []testcase{
+		{
+			name:    "no config",
+			layout:  []string{},
+			fromdir: "/",
+		},
+		{
+			name: "has config but no root config",
+			layout: []string{
+				"f:/config.tm:" + Block("generate_file",
+					Labels("test.txt"),
+					Str("content", "test"),
+				).String(),
+			},
+			fromdir: "/",
+		},
+		{
+			name: "has root config - find from root",
+			layout: []string{
+				"f:config.tm:" + Block("terramate",
+					Str("required_version", fmt.Sprintf("= %s", terramate.Version())),
+				).String(),
+			},
+			fromdir: "/",
+			want:    want{cfgpath: "/", found: true},
+		},
+		{
+			name: "has root config - find from nested subdir",
```

Comment:
this case was only tested indirectly in the e2e but it can go undetected if all tests use a git repo... as this is only used if it's not a repo and `terramate.required_version` is used.